test/unit/math/prim/fun/dot_self_test.cpp

```diff
@@ -56,21 +56,4 @@ TEST(MathMatrixPrimMat, dot_self) {
   Eigen::Matrix<double, 1, Eigen::Dynamic> rv3(3);
   rv3 << 2.0, 3.0, 4.0;
   EXPECT_NEAR(29.0, dot_self(rv3), 1E-12);
-
-  Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> m1(1, 1);
-  m1 << 2.0;
-  EXPECT_NEAR(4.0, dot_self(m1), 1E-12);
```

Comment:
Why gone?

Reply:
Oh because it only takes in vectors now? I wonder if this will break user code tho?

Reply:
Right. I should probably let this accept matrices too, even if matrix version is not exposed to stan language.